src/Badge.js

```diff
@@ -1,61 +1,61 @@
+import React from 'react'
 import PropTypes from 'prop-types'
 import styled from 'styled-components'
-import { space } from 'styled-system'
+import { type, space, color } from 'styled-system'
 import theme from './theme'
 
-function type(props) {
-  const badgeColors = {
-    green: {
-      backgroundColor: props.theme.colors.green,
-      color: props.theme.colors.white
-    },
-    lightGreen: {
-      backgroundColor: props.theme.colors.green0,
-      color: props.theme.colors.black
-    },
-    red: {
-      backgroundColor: props.theme.colors.red,
-      color: props.theme.colors.white
-    },
-    lightRed: {
-      backgroundColor: props.theme.colors.red0,
-      color: props.theme.colors.black
-    },
-    orange: {
-      backgroundColor: props.theme.colors.orange,
-      color: props.theme.colors.white
-    },
-    lightOrange: {
-      backgroundColor: props.theme.colors.orange0,
-      color: props.theme.colors.black
-    },
-    blue: {
-      backgroundColor: props.theme.colors.blue,
-      color: props.theme.colors.white
-    },
-    lightBlue: {
-      backgroundColor: props.theme.colors.blue0,
-      color: props.theme.colors.black
-    }
+const badgeColors = {
+  blue: {
+    bg: 'blue',
+    color: 'white'
+  },
+  lightBlue: {
+    bg: 'lightBlue',
+    color: 'darkBlue'
+  },
+  green: {
+    bg: 'green',
+    color: 'white'
+  },
+  lightGreen: {
+    bg: 'lightGreen',
+    color: 'darkGreen'
+  },
+  red: {
+    bg: 'red',
+    color: 'white'
+  },
+  lightRed: {
+    bg: 'lightRed',
+    color: 'darkRed'
+  },
+  orange: {
+    bg: 'orange',
+    color: 'text'
+  },
+  lightOrange: {
+    bg: 'lightOrange',
+    color: 'darkOrange'
   }
-
-  return badgeColors[props.bg]
 }
 
-const Badge = styled.div`
+const Base = styled.div`
   border-radius: 99999px;
   display: inline-block;
   font-size: ${props => props.theme.fontSizes[0]}px;
   font-weight: 600;
-  ${type} ${space};
+  text-transform: uppercase;
+  letter-spacing: 0.64px;
+  ${type} ${space} ${color};
 `
 
-Badge.defaultProps = {
-  px: 2,
-  py: 1,
-  theme: theme
+const Badge = props => {
+  const badgeColor = badgeColors[props.bg] || {}
```

Comment:
I like that this API sticks to the current convention, but it is probably worth discussing whether or not we want a color-combination-specific prop to handle both foreground and background colors